inference-engine/src/mkldnn_plugin/nodes/mkldnn_adaptive_pooling.cpp

```diff
@@ -0,0 +1,363 @@
+        IE_THROW() << errorPrefix << "has incorrect number of output edges: " << getChildEdges().size();
+    if (getChildEdges().size() != (algorithm == AdaptivePoolingMax ? 2 : 1))
+        IE_THROW() << errorPrefix << "has incorrect number of output edges: " << getParentEdges().size();
+
+    inputPrecision = getOriginalInputPrecisionAtPort(0);
+    outputPrecision = getOriginalOutputPrecisionAtPort(0);
+
+    auto parentDims = getParentEdgeAt(0)->getDims();
+    auto childDims = getChildEdgeAt(0)->getDims();
+
+    spatialDimsCount = parentDims.ndims() - 2;
+    if (spatialDimsCount != 1 && spatialDimsCount != 2 && spatialDimsCount != 3) {
+        IE_THROW() << errorPrefix << "doesn't support 0th input with rank: " << getParentEdgeAt(0)->getDims().ndims();
+    }
+
+    if (getParentEdgeAt(1)->getDims().ndims() != 1) {
+        IE_THROW() << errorPrefix << "doesn't support 1st input with rank: " << getParentEdgeAt(1)->getDims().ndims();
+    }
+
+    if (getChildEdgeAt(0)->getDims().ndims() != getParentEdgeAt(0)->getDims().ndims()) {
+        IE_THROW() << errorPrefix << "must keep data rank";
+    }
+}
+
+void MKLDNNAdaPoolingNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+    // we supports only floating-point types
+    if ((inputPrecision != Precision::FP32 && inputPrecision != Precision::BF16) ||
+        (outputPrecision != Precision::FP32 && outputPrecision != Precision::BF16))
+      outputPrecision = inputPrecision = Precision::FP32;
+
+    if (!mayiuse(avx512_core)) {
+        if (outputPrecision == Precision::BF16 || inputPrecision == Precision::BF16)
+            outputPrecision = inputPrecision = Precision::FP32;
+    }
+
+    auto inputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(inputPrecision);
+    auto outputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(outputPrecision);
+
+    InferenceEngine::LayerConfig config;
+    config.dynBatchSupport = false;
+    config.inConfs.resize(2);
+    config.outConfs.resize((algorithm == Algorithm::AdaptivePoolingAvg ? 1 : 2));
+
+    for (auto fmt : getAvailableFormatsForDims(getParentEdgeAt(0)->getDims())) {
+        config.inConfs[0].desc = MKLDNNMemoryDesc(getParentEdgeAt(0)->getDims(), inputDataType, fmt);
+        config.inConfs[1].desc = MKLDNNMemoryDesc(getParentEdgeAt(1)->getDims(), memory::data_type::s32, memory::format_tag::x);
+        config.outConfs[0].desc = MKLDNNMemoryDesc(getChildEdgeAt(0)->getDims(), outputDataType, fmt);
+        if (algorithm == Algorithm::AdaptivePoolingMax) {
+            memory::format_tag indexFmt;
+            switch (spatialDimsCount) {
+                case 1:
+                    indexFmt =  memory::format_tag::ncw;
+                    break;
+                case 2:
+                    indexFmt = memory::format_tag::nchw;
+                    break;
+                case 3:
+                default:
+                    indexFmt = memory::format_tag::ncdhw;
```

Comment:
Why ncdhw is default? Here should be a reaction to unexpected behavior.

Reply:
Done